### PR TITLE
Merge in 1.4.2 fixes - verified 

### PR DESCRIPTION
**Pull request by JacobGros**
Note: In draft until testing is complete 
* Add permissions for SC removal. (#431)

* Bump_up_csm-operator_version_to_1.4.1 (#434)

* Patch version support (#433)

* Adding support of v2.9.1 for Pmax, Pscale, PFlex and unity

* Updating module config version for Pmax, Pscale, PFlex and unity

* Update storage_csm_powerflex.yaml

* version changed to 2.9.1

* version changed to 2.9.1

* Enabling github actions for all branch

* updated version to 2.9.1 for unity, powerflex, powerscale and powermax

* Adding the support of PStore v2.9.1

* Adding the support of PFlex v2.9.1

* Version bumpup replication 1.7.1

* Bump up version for authorization, resiliency

* update csi-metadata-retriever version to 1.6.1

* update reverseproxy version to 2.8.1

---------

* Change namespace name to match documentation. (#437)

* go-yaml version update (#438)

* go-yaml version update

* update go-yaml package to latest

* remove extra verbs

* remove duplicate verbs and add correct test files for resiliency UT

---------

* [bug-1120]: Update Authorization sidecar to use insecure flag (#442)

* fix auth cert

* remove constant

* remove duplicate verbs

* reverting #426 changes (#444)

* Update final manifests for posting of Operator 1.4.1 (#445)

* KRV-20573/csm status (#446)

* add module status

* Adding CSM Module status

* Addressing the comments for PR

* Making code reusable

* Fix formatting issue

* Resolving gosec issue

* Final manifest update and apex client (#447)

* Add csm watch permissions. (#448)

* Merged CSM Module Status

* Fix App-Mob node error (#457)

* Bump up CSM Operator version to 1.4.2 (#458)

* update csm operator version to 1.4.2

* bump-up-op-version-1.4.2

* updated version in operator.yaml

* adding missing arg to fix e2e

* update authorization e2e (#459)

* skip cert validation in e2e (#460)

* Fixing FSGroupPolicy issue (#463)

* Fixing FSGroupPolicy issue

* Add unit test

---------

* Fix for controller race, daemon status, and obs namespace  (#464)

* test controller state calc

* add debug logs

* check for driver

* add daemonnset update

* adds ns fix for when pod ns dis not csm ns

* add ns check for daemon and deployment as well

* fix typo

* fix typo

* explicitly return nil

* add missing replace statements

---------

* Auth proxy status fix (#465)

* actions

* working on auth proxy status check

* finished status check

* fix 1

* fix or

* code compiles

* keep auth proxy out of daemonset check

* trying to figure out deployment name

* formatting

* fix auth name in if condition

* no daemonset

* fix

* added log statements

* added another log statement

* fix log statement

* logs logs logs

* compile

* more logs

* add logs and try updating status from handlesuccess

* remove UpdateStatus call for now

* adding csm label to auth

* remove extraneous string replacement

* removing commented code

* Update actions.yml

* fix merge issue

* fix fmt

* fix pscale + obs issue  (#467)

* fix some issues

* fiix label issue, add debug logs

* add more debug

* more  debugg lines

* fix pscale issue

* remmove debug logs

* gofmt

* set available to 0 en deployment is not checked

* fix if check

---------

* update final manifests (#468)

* fix vetting issues (#477)

---------

# Description
Merges in the fixes made for 1.4.2 into main branch 

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/1146 |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] E2E tests ran
- [x] Unit tests ran

### CODE REVIEW

**Comment by jooseppi-luna** [comment]
Just as an update, ran all PFlex E2E tests and only failed on one of the last app mobility scenarios because a docker image pull of an artifactory image was acting up (not operator related at all). Did find a weird edge case with PowerScale where the controller failed due to the csi-metadata-retriver v1.7.0 image not existing yet, but the csm status was marked as successful. I don't think this is a regression being introduced in this PR, so I will approve the PR.

```
[root@master-1-Zaglt7mQUY8Wg observability]# kubectl get pods -n isilon
NAME                                 READY   STATUS    RESTARTS       AGE
isilon-controller-67c5c64b86-bptxg   1/6     Error     12 (14s ago)   2m52s
isilon-node-2h9bt                    2/2     Running   0              2m52s
isilon-node-mtdmd                    2/2     Running   0              2m52s
[root@master-1-Zaglt7mQUY8Wg observability]# kubectl get csm -A
NAMESPACE   NAME     CREATIONTIME   CSIDRIVERTYPE   CONFIGVERSION   STATE
isilon      isilon   2m55s          isilon          v2.10.0         Succeeded
[root@master-1-Zaglt7mQUY8Wg observability]#
```

**Review by jooseppi-luna — APPROVED** [review]

File: controllers/csm_controller.go [hunk 1]
```diff
@@ -92,7 +92,7 @@ const (
 	CSMFinalizerName = "finalizer.dell.emc.com"
 
 	// CSMVersion -
-	CSMVersion = "v1.8.0"
+	CSMVersion = "v1.9.2"
```
Comment (jooseppi-luna) [inline controllers/csm_controller.go:95]: Should probably be 1.10.0?

File: bundle/manifests/dell-csm-operator.clusterserviceversion.yaml [hunk 2]
```diff
@@ -3575,5 +3575,5 @@ spec:
   - image: docker.io/dellemc/connectivity-cert-persister-k8s:0.7.0
     name: cert-persister
   skips:
-    - dell-csm-operator.v1.4.0
-  version: 1.4.1
+  - dell-csm-operator.v1.4.0
+  version: 1.4.2
```
Comment (jooseppi-luna) [inline bundle/manifests/dell-csm-operator.clusterserviceversion.yaml:3579]: why is this 1.4.2?

File: bundle/manifests/dell-csm-operator.clusterserviceversion.yaml [hunk 3]
```diff
@@ -3520,7 +3520,7 @@ spec:
     name: Dell Technologies
     url: https://github.com/dell/csm-operator
   relatedImages:
-  - image: docker.io/dellemc/dell-csm-operator:v1.4.1
+  - image: docker.io/dellemc/dell-csm-operator:v1.4.2
```
Comment (jooseppi-luna) [inline bundle/manifests/dell-csm-operator.clusterserviceversion.yaml:3523]: why is this 1.4.2?

File: controllers/csm_controller.go [hunk 4]
```diff
@@ -418,7 +421,11 @@ func (r *ContainerStorageModuleReconciler) handlePodsUpdate(_ interface{}, obj i
 
 	p, _ := obj.(*corev1.Pod)
 	name := p.GetLabels()[constants.CsmLabel]
-	ns := p.Namespace
+	//if this pod is an obs. pod, namespace might not match csm namespace
```
Comment (jooseppi-luna) [inline controllers/csm_controller.go:424]: add space at beginning of comment

File: pkg/utils/status.go [hunk 5]
```diff
@@ -320,27 +382,39 @@ func getDaemonSetStatus(ctx context.Context, instance *csmv1.ContainerStorageMod
 func calculateState(ctx context.Context, instance *csmv1.ContainerStorageModule, r ReconcileCSM, newStatus *csmv1.ContainerStorageModuleStatus) (bool, error) {
 	log := logger.GetLogger(ctx)
 	running := true
-	var err error
+	var err error = nil
```
Comment (jooseppi-luna) [inline pkg/utils/status.go:385]: drop ` = nil` part

File: pkg/utils/status.go [hunk 6]
```diff
@@ -544,18 +618,21 @@ func HandleSuccess(ctx context.Context, instance *csmv1.ContainerStorageModule,
...
 	log.Infow("HandleSuccess Driver state ", "newStatus.State", newStatus.State)
-	if newStatus.State == constants.Running {
+	if newStatus.State == constants.Succeeded {
 		// If previously we were in running state
-		if oldStatus.State == constants.Running {
-			log.Info("HandleSuccess Driver state didn't change from Running")
+		if oldStatus.State == constants.Succeeded {
+			log.Info("HandleSuccess Driver state didn't change from Succeeded")
+		} else {
+			log.Info("HandleSuccess Driver stat changed to Succeeded")
```
Comment (jooseppi-luna) [inline pkg/utils/status.go:635]: *state

**Review by jooseppi-luna — CHANGES_REQUESTED** [review]
can we run gofumpt on csm_controller.go and status.go?

**Comment by JacobGros** [comment]
Closing and re-opening due to verified issue
